bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -1335,4 +1341,62 @@ public void updated(Dictionary<String, ?> properties) throws ConfigurationExcept
         }
     }
 
+    /**
+     * The method checks configuration of module if it is available and if all required properties are set.
+     * If some property is tries to replace it with its default value.
+     *
+     * @param module checked module
+     */
+    private void checkConfigurations(Module module) {
+        String type = module.getTypeUID();
+        if (mtManager != null) {
+            Map<String, Object> mConfig = module.getConfiguration();
+            if (mConfig == null) {
+                mConfig = new HashMap<>(11);
+            }
+            ModuleType mt = mtManager.get(type);
+            if (mt != null) {
+                List<ConfigDescriptionParameter> configDescriptions = mt.getConfigurationDescription();
+                for (ConfigDescriptionParameter cftDesc : configDescriptions) {
+                    String parameterName = cftDesc.getName();
+                    if (mConfig.get(parameterName) == null) {
+                        String strValue = cftDesc.getDefault();
+                        if (strValue != null) {
+                            Type t = cftDesc.getType();
+                            Object defValue = t != Type.TEXT ? getDefaultValue(t, strValue) : strValue;
+                            mConfig.put(parameterName, defValue);
+                        } else {
+                            if (cftDesc.isRequired()) {
+                                throw new IllegalArgumentException(
+                                        "Missing required parameter: " + parameterName + " of type " + type);
+                            }
+                        }
+                    }
+                }
+            }
+            module.setConfiguration(mConfig);
+        } else {
+            logger.warn("Can't get module type definition for:" + type + ". Missing ModuleTypeManager");
+        }
+    }
+
+    /**
+     * The method parses string presentation of default value
+     *
+     * @param type type of default object
+     * @param value string presentation of default object
+     * @return default value
+     */
+    private Object getDefaultValue(Type type, String value) {
+        switch (type) {
+            case BOOLEAN:
+                return Boolean.valueOf(value);
+            case INTEGER:
+                return Integer.valueOf(value);
+            case DECIMAL:
+                return Double.valueOf(value);
```

Comment:
I doubt that we have defined it this way, since double doesn't allow precise calculations and in the rest of ESH we are using BigDecimal for numbers. Specifically, when receiving configurations as JSON from the REST API, we use https://github.com/eclipse/smarthome/blob/master/bundles/io/org.eclipse.smarthome.io.rest/src/main/java/org/eclipse/smarthome/io/rest/ConfigUtil.java#L29.
I think it would make a lot of sense to apply the same logic for automation configurations. Note that GSON actually does not differentiate between int and double at all, so you will actually never get a configuration with int values from the parser.